File: ゼミ用/05製造/web/graph/request/DeleteTimeRequest.php
<?php
require_once dirname(__FILE__) . '/../common/Constant.php';

/**
 * 時系列削除処理の値を保持する
 */
class DeleteTimeRequest {
    private $timeId;
    private $userId;
    private $errorMsg = array();

    /**
     * 時系列削除処理の値を格納します
     */
    public function __construct($timeId, $userId) {
        $this->timeId = $timeId;
        $this->userId = $userId;
    }

    public function getTimeId() {
        return $this->timeId;
    }

    public function getUserId() {
        return $this->userId;
    }

    public function getErrorMsg() {
        return $this->errorMsg;
    }

    /**
     * バリデーションチェックをします
     */
    public function validation() {
        $validationFlg = false;

        try {
            // 入力項目チェック
            if (empty($this->timeId) || empty($this->userId)) {
                array_push($this->errorMsg, 'エラー ：必須項目が入力されていません');
                $validationFlg = true;
            }

            // 時系列IDの文字数チェック
            if (strlen($this->timeId) != Constant::TIME_ID_DIGIT + strlen(Constant::TIME_ID_STR)) {
                array_push($this->errorMsg, 'エラー ：時系列IDは' . Constant::TIME_ID_DIGIT + strlen(Constant::TIME_ID_STR) . '文字で入力してください');
                $validationFlg = true;
            }

            // ユーザIDの文字数チェック
            if (strlen($this->userId) != Constant::USER_ID_DIGIT + strlen(Constant::USER_ID_STR)) {
                array_push($this->errorMsg, 'エラー ：ユーザIDは' . Constant::USER_ID_DIGIT + strlen(Constant::USER_ID_STR) . '文字で入力してください');
                $validationFlg = true;
            }
        } catch (Exception $e) {
            $validationFlg = true;
        }

        return $validationFlg;
    }
}

File: ゼミ用/05製造/eeldev/eelPy/api/request/DeleteTimeRequest.py
from ..common.Constant import Constant


# /**
# * 時系列削除処理の値を保持する
# */
class DeleteTimeRequest:
    errorMsg = []

    # /**
    # * 時系列削除処理の値を格納します
    # */
    def __init__(self, timeId, userId):
        self.timeId = timeId
        self.userId = userId

    def getTimeId(self):
        return self.timeId

    def getUserId(self):
        return self.userId

    def getErrorMsg(self):
        return self.errorMsg

    # /**
    # * バリデーションチェックをします
    # */
    def validation(self):
        validationFlg = False

        try:
            # 入力項目チェック
            if (len(self.timeId) == 0 or len(self.userId) == 0):
                self.errorMsg.append('エラー ：必須項目が入力されていません')
                validationFlg = True

            # 時系列IDの文字数チェック
            if (Constant.TIME_ID_DIGIT + len(Constant.TIME_ID_STR) < len(self.timeId)):
                self.errorMsg.append(
                    'エラー ：時系列IDは' + Constant.TIME_ID_DIGIT + len(Constant.TIME_ID_STR) + '文字以内で入力してください')
                validationFlg = True

            # ユーザIDの文字数チェック
            if (Constant.USER_ID_DIGIT + len(Constant.USER_ID_STR) < len(self.userId)):
                self.errorMsg.append(
                    'エラー ：ユーザIDは' + Constant.USER_ID_DIGIT + len(Constant.USER_ID_STR) + '文字以内で入力してください')
                validationFlg = True

        except Exception as e:
            validationFlg = True

        return validationFlg
